guidance/d_star_lite/d_star_lite/d_star_lite_node.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+import rclpy
+from rclpy.node import Node
+import numpy as np
+from d_star_lite import DStarLite, DSLNode
+from vortex_msgs.srv import MissionPlanner, Waypoint
+
+class DStarLiteNode(Node):
+    """
+    A ROS2 node implementing the D* Lite algorithm.
+
+    The node offers a mission planner service to calculate the optimal waypoints, which are then sent to the waypoint service.
+    """
+    def __init__(self):
+        """
+        Initialize the DStarLiteNode, creating necessary services and clients
+        for mission planning and waypoint submission.
+        """
+        super().__init__('d_star_lite_node')
+        self.obs_srv = self.create_service(MissionPlanner, 'mission_planner', self.d_star_lite_callback)
+        self.wp_client = self.create_client(Waypoint, 'waypoint')
+        self.get_logger().info('D Star Lite Node has been started')
+        self.WP_float = []
+        
+
+    def d_star_lite_callback(self, request, response):
+        """
+        Callback for the mission planner service.
+
+        Args:
+            request: start and goal coordinates, and the obstacle coordinates
+            response: success flag
+
+        Returns:
+            The modified response object with success status
+        """
+        self.get_logger().info('D Star Lite Service has been called')
+        ox = request.ox
+        oy = request.oy
+        sx = request.sx
+        sy = request.sy
+        gx = request.gx
+        gy = request.gy
+        dsl = DStarLite(ox, oy)
+        dsl.dsl_main(DSLNode(sx, sy), DSLNode(gx, gy))
```

Comment:
This crashes when calling the mission planner service from the terminal with a single x,y for start, goal, and obstacle.

The error:
d_star_lite.py", line 292, in initialize
[d_star_lite_node.py-1]     self.rhs[self.goal.x][self.goal.y] = 0
[d_star_lite_node.py-1] IndexError: index 40 is out of bounds for axis 0 with size 0

Reply:
Fixed now